hal/src/stm32f2xx/i2c_hal.c

```diff
@@ -181,54 +188,99 @@ static void HAL_I2C_SoftwareReset(HAL_I2C_Interface i2c)
     i2cMap[i2c]->prevEnding = I2C_ENDING_UNKNOWN;
 }
 
-void HAL_I2C_Init(HAL_I2C_Interface i2c, void* reserved)
+static bool HAL_I2C_Config_Is_Valid(const HAL_I2C_Config* config) {
+    if ((config == NULL) ||
+            (config->rx_buffer == NULL ||
+             config->rx_buffer_size == 0 ||
+             config->tx_buffer == NULL ||
+             config->tx_buffer_size == 0))
+    {
```

Comment:
Nitpicking: move it to following the `if` expression.

Reply:
Fixed.